src/Compilers/VisualBasic/BasicAnalyzerDriver/VisualBasicDeclarationComputer.vb

```diff
@@ -100,62 +116,97 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                         ComputeDeclarationsCore(model, decl, shouldSkip, getSymbol, builder, newLevel, cancellationToken)
                     Next
+
+                    If Not t.Attributes.IsEmpty Then
+                        Dim attributes = GetAttributes(t.Attributes)
+                        builder.Add(GetDeclarationInfo(model, node, getSymbol, attributes, cancellationToken))
+                    End If
                     Return
                 Case Else
                     Dim typeBlock = TryCast(node, TypeBlockSyntax)
                     If typeBlock IsNot Nothing Then
                         For Each decl In typeBlock.Members
                             ComputeDeclarationsCore(model, decl, shouldSkip, getSymbol, builder, newLevel, cancellationToken)
                         Next
-                        builder.Add(GetDeclarationInfo(model, node, getSymbol, cancellationToken))
+                        Dim attributes = GetAttributes(typeBlock.BlockStatement.AttributeLists)
+                        builder.Add(GetDeclarationInfo(model, node, getSymbol, attributes, cancellationToken))
                         Return
                     End If
 
                     Dim typeStatement = TryCast(node, TypeStatementSyntax)
                     If typeStatement IsNot Nothing Then
-                        builder.Add(GetDeclarationInfo(model, node, getSymbol, cancellationToken))
+                        Dim attributes = GetAttributes(typeStatement.AttributeLists)
+                        builder.Add(GetDeclarationInfo(model, node, getSymbol, attributes, cancellationToken))
                         Return
                     End If
 
                     Dim methodBlock = TryCast(node, MethodBlockBaseSyntax)
                     If methodBlock IsNot Nothing Then
-                        Dim paramInitializers = GetParameterInitializers(methodBlock.BlockStatement.ParameterList)
-                        Dim codeBlocks = paramInitializers.Concat(methodBlock)
+                        Dim paramInitializers = GetParameterListInitializersAndAttributes(methodBlock.BlockStatement.ParameterList)
+                        Dim attributes = GetAttributes(methodBlock.BlockStatement.AttributeLists)
+                        Dim codeBlocks = paramInitializers.Concat(methodBlock).Concat(attributes)
                         builder.Add(GetDeclarationInfo(model, node, getSymbol, codeBlocks, cancellationToken))
                         Return
                     End If
 
                     Dim methodStatement = TryCast(node, MethodBaseSyntax)
                     If methodStatement IsNot Nothing Then
-                        Dim paramInitializers = GetParameterInitializers(methodStatement.ParameterList)
-                        builder.Add(GetDeclarationInfo(model, node, getSymbol, paramInitializers, cancellationToken))
+                        Dim paramInitializers = GetParameterListInitializersAndAttributes(methodStatement.ParameterList)
```

Comment:
>Dim paramInitializers = GetParameterListInitializersAndAttributes(methodStatement.ParameterList) [](start = 24, length = 96)

Are we handling attributes on return type?
```
    Function Test() As <System.Runtime.InteropServices.MarshalAs(System.Runtime.InteropServices.UnmanagedType.Currency)> Integer
        Return 0
    End Function
```

Reply:
@AlekseyTs Thanks, handled now.